src/openwhisk/static.js

```diff
@@ -266,7 +267,26 @@ function deliverPlain(owner, repo, ref, entry, root, esi = false) {
         'X-Static': 'Raw/Static',
       },
     };
-  }).catch(rqerror => error(rqerror.response.body.toString(), rqerror.statusCode));
+  }).catch((rqerror) => {
+    if (esi) {
+      // the ESI failed, so we simply fall back to the original URL
+      // the browser will fetch it again, so let's cache the 404
+      // for five minutes, in order to prevent the static function
+      // from being called too often
+      return {
+        statusCode: 404,
+        headers: {
+          'Content-Type': 'text/plain',
+          'Cache-Control': 's-max-age=300',
+        },
+        body: entry,
+      };
+    }
+    if (rqerror.statusCode === 404 || rqerror.statusCode === '404') {
+      return error(entry, rqerror.statusCode);
```

Comment:
This looks like an attack vector for reflected XSS: `error()` will set a `text/html` response and use the `message` as `body`...

Reply:
done.